q2_diversity/_beta/_method.py

```diff
@@ -63,6 +76,31 @@ def beta_phylogenetic(table: biom.Table, phylogeny: skbio.TreeNode,
     return results
 
 
+def beta_phylogenetic_alt(table: BIOMV210Format, phylogeny: NewickFormat,
+                          metric: str, n_jobs: int=1,
+                          variance_adjusted: bool=False,
+                          alpha=None,
+                          bypass_tips: bool=False) -> skbio.DistanceMatrix:
+
+    metrics = phylogenetic_metrics_alt_dict()
+
+    if metric not in metrics:
+        raise ValueError("Unknown metric: %s" % metric)
+
+    if alpha is not None and metric != 'generalized_unifrac':
+        raise ValueError('The alpha parameter is only allowed when the choice'
+                         ' of metric is generalized_unifrac')
+
+    if metric == 'generalized_unifrac':
+        f = partial(metrics[metric], alpha=alpha)
+    else:
+        f = metrics[metric]
+
+    # unifrac processes tables and trees should be filenames
+    return f(str(table), str(phylogeny), threads=n_jobs,
```

Comment:
when `n_jobs` is greater than logical cores, we get a segfault (when we're lucky). We should just guard that value to be at or less than cores present on the host.

Reply:
`psutil` to the rescure 💪